File: python/KartaPracy6a.py
# zad 1

# x = int(input())
# suma = 0
# while x > 0:
#     suma += x % 10
#     x = x // 10
# print(suma)


# zad 2


#
#
# def pierwsza(n):
#   if n < 2:
#     return False
#   elif n == 2:
#     return True
#   for i in range(2, n):
#     if n % i == 0:
#       return False
#   return True
#
# n = int(input())
# print(pierwsza(n))



# zad 3

# n = int(input())
# suma = 0
#
# for i in range(1, n):
#     if i%n == 0:
#         suma += i
# if suma == n:
#     print("Liczba jest doskonała")
# else:
#     print("Liczba nie jest doskonała")


# zad 4
#
def nwd(a, b):
    while b > 0:
        a, b = b, a%b
    return a


# a, b = int(input()), int(input())


# if nwd(a, b) == 1:
#     print("Liczby są względnie pierwsze")
# else:
#     print("Liczby nie są względnie pierwsze")


# zad 5


# n = int(input())
# for i in range(10,20):
#     if nwd(n, i) == 1:
#         print(i)


# zad 6

# a, b = map(int, input().split("/"))
# print(f"{a//nwd(a, b)}/{b//nwd(a, b)}")


# zad 7

# a, b = map(int, input().split("/"))
# c = a // b
# print(f"{c}  {(a%b)//nwd(a, b)}/{b//nwd(a, b)}")

# zad 8



# for i in range(1, 10000):
#     a = 0
#     for j in range(1, i):
#         if i % j == 0:
#             a += j
#     b = 0
#     for j in range(1, a):
#         if a % j == 0:
#             b += j
#     if i == b and i != a:
#         print(f"{i} i {a} są liczbami zaprzyjaźnionymi")
# exit(2137)




# zad 9



# for i in range(10,100):
#     if pierwsza(i):
#         for j in range(10,100):
#             if pierwsza(j):
#                 print(f"{i}*{j} = {i*j}")



# zad 10/1


# n = int(input())
#
# if pierwsza(n):
#     if pierwsza(n+2):
#         print(n, "jest liczbą bliźniaczą, bo ma bliźniaka", n+2)
#     else:
#         print(n, "nie jest liczbą bliźniaczą")
# else:
#     print(n, "nie jest liczbą pierwszą")


# zad 10/2


# for i in range(1000):
#     if pierwsza(i) and pierwsza(i+2):
#         print(i, "ma bliźniaka", i+2)

File: c#/KartaPracy6a.cs
using System;
using System.Net;
using System.Runtime.InteropServices;

// zad 1

int n = int.Parse(Console.ReadLine());

int suma = 0;
while (n > 0)
{
    suma += n % 10;
    n = n / 10;
}
Console.WriteLine(suma);


// zad 2




int n = int.Parse(Console.ReadLine());

if (n < 2)
{
    Console.WriteLine("Liczba nie jest pierwsza");
}
else if (n == 2) 
{
    Console.WriteLine("Liczba jest pierwsza");
    
}

for (int i = 2; i < n; i++)
{
    if (n % i == 0)
    {
        Console.WriteLine("Liczba nie jest pierwsza");
        break;
    }
    else
    {
        Console.WriteLine("Liczba jest pierwsza");
        break;
    }
}


// zad 3




int n = int.Parse(Console.ReadLine());
int suma = 0;
for (int i = 1; i < n; i++)
{
    if (n % i == 0)
    {
        suma += i;
    }
}

if (suma == n)
{
    Console.WriteLine("Liczba jest doskonała");
}
else
{
    Console.WriteLine("Liczba nie jest doskonała");
}



// zad 4





int x = int.Parse(Console.ReadLine());
int y = int.Parse(Console.ReadLine());

int NWD(int a, int b)
{
    while (b != 0)
    {
        int temp = a % b;
        a = b;
        b = temp;
    }

    return a;
}

if (NWD(x, y) == 1)
{
    Console.WriteLine("Liczby są względnie pierwsze");
}
else
{
    Console.WriteLine("Liczby nie są względnie pierwsze");
}


//zad 5



int n = int.Parse(Console.ReadLine());

for (int i = 10; i < 20; i++)
{
    if (NWD(n, i) == 1)
    {
        Console.WriteLine(i);
    }
}


// zad 6

Console.Write("Podaj licznik: ");
int a = int.Parse(Console.ReadLine());
Console.Write("Podaj mianownik: ");
int b = int.Parse(Console.ReadLine());

a = a / NWD(a, b);
b = b / NWD(a, b);

Console.WriteLine($"Skrócony ułamek: {a}/{b}");


// zad 7


Console.Write("Podaj licznik: ");
int a = int.Parse(Console.ReadLine());
Console.Write("Podaj mianownik: ");
int b = int.Parse(Console.ReadLine());

int c = a / b;
Console.WriteLine($"{c}  {(a%b)/NWD(a, b)}/{b/NWD(a, b)}");



//zad 8




for (int i = 2; i < 10000; i++)
{
    int a = 0;
    
    for (int j = 1; j < i; j++)
    {
        if (i%j==0)
        {
            a += j;
        }
    }

    int b = 0;
    
    for (int j = 1; j < a; j++)
    {
        if (a % j == 0)
        {
            b += j;
        }
    }

    if (i == b && i != a)
    {
        Console.WriteLine($"{i} i {a} są liczbami zaprzyjaźnionymi");
    }
}





// zad 9



int pierwsza(int n)
{
    if (n < 2)
    {
        return 0;
    }
    else if (n == 2)
    {
        return 1;
    }

    for (int i = 2; i < n; i++)
    {
        if (n % i == 0)
        {
            return 0;
        }
    }
    return 1;
}

for (int i = 10; i < 100; i++)
{
    if (pierwsza(i) == 1)
    {
        for (int j = 10; j < 100; j++)
        {
            if (pierwsza(j) == 1)
            {
                Console.WriteLine($"{i*j} = {i}*{j}");
            }
        }
    }
}






// zad 10




int n = int.Parse(Console.ReadLine());

if (pierwsza(n) == 1 && pierwsza(n + 2) == 1)
{
    Console.WriteLine($"Tak, liczba {n} ma bliźniaczą liczbę {n+2}");
}
else
{
    Console.WriteLine("Nie, a liczb anie ma bliźniaka lub nie jest pierwsza");
}

